### PR TITLE
Adding --camera-direction and --camera-zoom-factor

### LINKED ISSUE #45: Better default camera location
https://gitlab.kitware.com/f3d/f3d/-/issues/232

### PR DESCRIPTION
**Pull request by mwestphal**
* Add `--camera-direction` option
 * Add `--camera-zoom-factor` option (need vtk 9.3)
 * Use `camera-direction` in default config file for better default camera direction (open to changes on this one)
 * Fix #45

### CODE REVIEW

File: python/testing/TestPythonCamera.py [hunk 1]
```diff
@@ -38,5 +38,5 @@
 camera.pitch(angle)
 
 camera.setCurrentAsDefault()
-camera.resetToBounds()
+camera.resetToBounds(0.9)
```
Comment (mwestphal) [inline python/testing/TestPythonCamera.py:41]: can we have default values in pybind11 ?

Comment (snoyer) [reply python/testing/TestPythonCamera.py:41]: `.def("resetToBounds", &f3d::camera::resetToBounds, py::arg("zoomFactor") = 0.9);` should work

File: application/F3DStarter.cxx [hunk 2]
```diff
@@ -47,6 +47,33 @@ class F3DStarter::F3DInternals
...
+      f3d::point3_t pos;
+      cam.getFocalPoint(foc);
+      for (int i = 0; i < 3; i++)
+      {
+        pos[i] = foc[i] - dir[i];
+      }
+      cam.setPosition(pos);
+      reset = true;
+    }
+    if (appOptions.CameraPosition.size() != 3 && appOptions.CameraZoomFactor != 0)
```
Comment (Meakk) [inline application/F3DStarter.cxx:67]: what happens if `CameraZoomFactor` is negative?

File: doc/user/LIMITATIONS_AND_TROUBLESHOOTING.md [hunk 3]
```diff
@@ -2,12 +2,13 @@
 
 Here is a non exhaustive list of F3D limitations:
 
-* No support for specifying manual lighting in the default scene.
+* No support for specifying manual lighting in the default scene apart using `--light-intensity` option.
```
Comment (Meakk) [inline doc/user/LIMITATIONS_AND_TROUBLESHOOTING.md:5]:
```suggestion
* No support for specifying manual lighting in the default scene apart from using `--light-intensity` option.
```

File: doc/user/OPTIONS.md [hunk 4]
```diff
@@ -85,12 +85,14 @@ Options|Default|Description
 
 Options|Default|Description
 ------|------|------
-\-\-camera-position=\<X,Y,Z\>||Set the camera position.
+\-\-camera-position=\<X,Y,Z\>||Set the camera position, override --camera-direction and camera-zoom-factor.
```
Comment (Meakk) [inline doc/user/OPTIONS.md:88]:
```suggestion
\-\-camera-position=\<X,Y,Z\>||Set the camera position, overrides --camera-direction and camera-zoom-factor.
```

File: application/F3DStarter.cxx (hunk 2 above)
Comment (mwestphal) [reply application/F3DStarter.cxx:67]: it is ignored by VTK, but lets ignore it at F3D level instead.

File: application/F3DOptionsParser.cxx [hunk 5]
```diff
@@ -337,12 +337,14 @@ void ConfigurationOptions::GetOptions(F3DAppOptions& appOptions, f3d::options& o
...
     this->DeclareOption(grpCamera, "camera-view-angle", "", "Camera view angle (non-zero, in degrees)", appOptions.CameraViewAngle, LocalHasDefaultNo, MayHaveConfig::YES, "<angle>");
-    this->DeclareOption(grpCamera, "camera-azimuth-angle", "", "Camera azimuth angle (in degrees)", appOptions.CameraAzimuthAngle, HasDefault::YES, MayHaveConfig::YES, "<angle>");
-    this->DeclareOption(grpCamera, "camera-elevation-angle", "", "Camera elevation angle (in degrees)", appOptions.CameraElevationAngle, HasDefault::YES, MayHaveConfig::YES, "<angle>");
+    this->DeclareOption(grpCamera, "camera-direction", "", "Camera direction", appOptions.CameraDirection, HasDefault::YES, MayHaveConfig::YES, "<X,Y,Z>");
+    this->DeclareOption(grpCamera, "camera-zoom-factor", "", "Camera zoom factor (non-zero)", appOptions.CameraZoomFactor, HasDefault::YES, MayHaveConfig::YES, "<factor>");
+    this->DeclareOption(grpCamera, "camera-azimuth-angle", "", "Camera azimuth angle (in degrees), performed after other camera options except elevation", appOptions.CameraAzimuthAngle, HasDefault::YES, MayHaveConfig::YES, "<angle>");
```
Comment (snoyer) [inline application/F3DOptionsParser.cxx:346]: Is the `elevation`,`azimuth` order significant? if not:
```suggestion
    this->DeclareOption(grpCamera, "camera-azimuth-angle", "", "Camera azimuth angle (in degrees), performed after other camera options", appOptions.CameraAzimuthAngle, HasDefault::YES, MayHaveConfig::YES, "<angle>");
```

Comment (mwestphal) [reply application/F3DOptionsParser.cxx:346]: good point, it doesnt

**Review by Meakk — APPROVED** [review]

**Pull request merged**